src/consensus/proposer.rs

```diff
@@ -492,8 +493,16 @@ impl Proposer for BlockProposer {
         round: Round,
         timeout: Duration,
     ) -> FullProposal {
-        let timestamp = FarcasterTime::current();
-        let version = EngineVersion::version_for(&timestamp, self.network);
+        let mempool_timeout = Duration::from_millis(200);
+        let messages = self
+            .engine
+            .mempool_poller
+            .pull_messages(mempool_timeout)
```

Comment:
We also need to duplicate onchain events to shard 0 in the mempool right? I don't think it has a concept of shard 0 right now?

Reply:
Yeah, I didn't want to add that here because I don't want onchain events to get into shard 0 blocks yet. I wanted to test that the system works with just heartbeats first. We can route onchain events to shard 0 without an engine version upgrade later once we're confident the system works. The reason I added support for onchain events in the engine is so that I can test all the code that deals with the trie. I wanted to see that the shard root updated as expected. Added a comment to remember to route onchain events to shard 0.